packages/eas-cli/src/credentials/ios/actions/new/RemoveDistributionCertificate.ts

```diff
@@ -0,0 +1,81 @@
+import { AppleDistributionCertificateFragment } from '../../../../graphql/generated';
+import log from '../../../../log';
+import { confirmAsync } from '../../../../prompts';
+import { Account } from '../../../../user/Account';
+import { Action, CredentialsManager } from '../../../CredentialsManager';
+import { Context } from '../../../context';
+import { selectDistributionCertificateWithDependenciesAsync } from './DistributionCertificateUtils';
+import { RemoveProvisioningProfiles } from './RemoveProvisioningProfile';
+
+export class SelectAndRemoveDistributionCertificate implements Action {
+  constructor(private account: Account) {}
+
+  async runAsync(_manager: CredentialsManager, ctx: Context): Promise<void> {
+    const selected = await selectDistributionCertificateWithDependenciesAsync(ctx, this.account);
+    if (selected) {
+      await new RemoveDistributionCertificate(this.account, selected).runAsync(ctx);
+      log.succeed('Removed distribution certificate');
+      log.newLine();
+    }
+  }
+}
+
+export class RemoveDistributionCertificate {
+  constructor(
+    private account: Account,
+    private distributionCertificate: AppleDistributionCertificateFragment
+  ) {}
+
+  public async runAsync(ctx: Context): Promise<void> {
+    const apps = this.distributionCertificate.iosAppBuildCredentialsList.map(
+      buildCredentials => buildCredentials.iosAppCredentials.app
+    );
+    if (apps.length !== 0 && !ctx.nonInteractive) {
```

Comment:
This is just an idea. Maybe we should consider throwing an error if the certificate is used by at least one app and we're in non-interactive mode.